addons/crm/models/crm_team_member.py

```diff
@@ -209,4 +209,4 @@ def _get_assignment_quota(self, work_days=1):
         assign_ratio = work_days / 30.0
         to_assign = self.assignment_max * assign_ratio
         compensation = max(0, self.assignment_max - (self.lead_month_count + to_assign)) * 0.2
-        return round(to_assign + compensation)
+        return max(1, round(to_assign + compensation))
```

Comment:
I think this might make more sense on `to_assign` since that's really the problematic part of the computation. And `compensation` will adjust for it anyway.

Reply:
Hey @thiryr!

1) The ValueError gets generated by the `random` python library in this line https://github.com/python/cpython/blob/634cb61909b1392f0eef3e749931777adb54f16e/Lib/random.py#L478 which gets called at https://github.com/odoo/odoo/blob/d7d0527a9c4a701c23a5f44b60c9eae7e13126fb/addons/crm/models/crm_team_member.py#L161

2) It should also happen in 15.0 since the code is the same but not in 14.0, as far as I checked.

3) There is an existing test that should test if the updated method functions as expected https://github.com/odoo/odoo/blob/5afcd06168f1a9802b6ba4eedc29cdbcb637c654/addons/crm/tests/test_crm_lead_assignment.py#L400

Also, I made the change you suggested. It does make more sense to rectify the `to_assign` instead of the `to_assign + compensation`.

Reply:
Change applied!